### PR TITLE
Optimized verify_cell_kzg_proof_batch by adding #[cfg(feature = "parallel")] in the function itself and also in some helper functions

### PR DESCRIPTION
**Pull request by Myshk-IN**
Optimized verify_cell_kzg_proof_batch by adding #[cfg(feature = "parallel")] in the function itself and also in some helper functions.

Please, check if everything works as it should. If I am right, these changes should speed up verify_cell_kzg_proof_batch function and should not break any code.

### CODE REVIEW

File: kzg/src/eip_7594.rs [hunk 1]
```diff
@@ -153,20 +156,45 @@ fn compute_fk20_proofs<
...
-        );
+    #[cfg(feature = "parallel")]
+    {
+        h_ext_fft.par_iter_mut().enumerate().for_each(|(i, h_ext_fft_elem)| {
+            *h_ext_fft_elem = TG1::g1_lincomb(
+                s.get_x_ext_fft_column(i),
+                &coeffs[i],
+                FIELD_ELEMENTS_PER_CELL,
+                None,
+            );
```
Comment (ArtiomTr) [inline kzg/src/eip_7594.rs]: g1_lincomb is already parallelized, but due to small MSM size this doesn't give any performance penalty, but still not a good thing. Should probably be removed for now, later we'll add g1_lincomb_batch trait, that will run `N` parallel small MSMs, or something like that.

File: kzg/src/eip_7594.rs [hunk 2]
```diff
@@ -591,19 +619,50 @@ fn compute_r_powers_for_verify_cell_kzg_proof_batch<TG1: G1, TFr: Fr>(
...
+                local_weights
+            })
+            .collect();
+
+        for local_weights in intermediate_weights {
+            for (i, weight) in local_weights.into_iter().enumerate() {
+                commitment_weights[i] = commitment_weights[i].add(&weight);
+            }
+        }
+    }
```
Comment (ArtiomTr) [inline kzg/src/eip_7594.rs]: Very clever! Have you tried picking chunk size, other than `64`? My intuition would be that picking `r_powers.len() / available_parallelism()` will show good results. Also, have you considered parallelizing summation part?

File: kzg/src/eip_7594.rs [hunk 3]
```diff
@@ -153,20 +156,45 @@ fn compute_fk20_proofs<
...
+        h.par_iter_mut().take(k2).skip(k).for_each(|h_elem| {
+            *h_elem = TG1::identity();
+        });
+    }
+
+    #[cfg(not(feature = "parallel"))]
+    {
+        for h_elem in h.iter_mut().take(k2).skip(k) {
+            *h_elem = TG1::identity();
+        }
```
Comment (ArtiomTr) [inline kzg/src/eip_7594.rs]: This kind of code could be a bit simplified with `kzg::cfg_into_iter` macro (or similar one), so you don't need 2 different implementations, and could just do:
```rs
cfg_iter_mut!(h).take(k2).skip(k).for_each(|h_elem| {
    *h_elem = TG1::identity();
})
```

(`cfg_iter_mut` macro calls `iter_mut`/`par_iter_mut` depending on `parallel` feature)

**Review by ArtiomTr — CHANGES_REQUESTED** [review]
Looks good, minor changes requested 👍

File: kzg/src/eip_7594.rs (hunk 3 above)
Comment (Myshk-IN) [reply kzg/src/eip_7594.rs]: Ok, it is now changed

File: kzg/src/eip_7594.rs (hunk 2 above)
Comment (Myshk-IN) [reply kzg/src/eip_7594.rs]: I encountered some bugs, so I decided not to parallelize the summation part

File: kzg/src/eip_7594.rs (hunk 1 above)
Comment (Myshk-IN) [reply kzg/src/eip_7594.rs]: Got it, parallelization is removed.

**Review by ArtiomTr — APPROVED** [review]
Perfect 💯

**Pull request merged**